imgui.cpp

```diff
@@ -6248,6 +6266,72 @@ void ImGui::PopTextWrapPos()
     window->DC.TextWrapPos = window->DC.TextWrapPosStack.empty() ? -1.0f : window->DC.TextWrapPosStack.back();
 }
 
+// These functions are quite slow because they are 'correct', but currently they are only used for conversions within ImGui itself in a one-off manner
+float ImGui::SRGBToLinear(float in)
+{
+    if (in <= 0.04045f)
+        return in / 12.92f;
+    else
+        return ImPow((in + 0.055f) / 1.055f, 2.4f);
+}
+
+float ImGui::LinearToSRGB(float in)
+{
+    if (in <= 0.0031308f)
+        return in * 12.92f;
+    else
+        return 1.055f * ImPow(in, 1.0f / 2.4f) - 0.055f;
+}
+
+ImVec4 ImGui::SRGBToLinear(ImVec4 col)
+{
+    col.x = SRGBToLinear(col.x);
+    col.y = SRGBToLinear(col.y);
+    col.z = SRGBToLinear(col.z);
+    // Alpha component is already linear
+
+    return col;
+}
+
+ImVec4 ImGui::LinearToSRGB(ImVec4 col)
+{
+    col.x = LinearToSRGB(col.x);
+    col.y = LinearToSRGB(col.y);
+    col.z = LinearToSRGB(col.z);
+    // Alpha component is already linear
+
+    return col;
+}
+
+void ImGui::ForceStyleColorSpaceConversion(ImGuiStyle* style, bool to_linear)
```

Comment:
It would be good to keep track of the current color space of the style. That would allow this function to do the right thing even if style already was in the requested color space. Currently, you need to externally figure out if calling this function will work, otherwise style colors will be in broken color space.

Reply:
Hiyas, thanks for the comment. You're right in that styles could document whether or not they are linear colour, so that users could use this function a little more freely in more general contexts, and convert their existing styles safely and on the fly

The original design for this PR was that there'd be no mix and matching of colour spaces however - once ImGui was in linear colour mode, absolutely everything would be in linear colour - so this function is essentially a bit of a hack for the built in imgui styles

That said this PR seems to have become a tad stuck, so I'm not really sure where I'm going with this!

Reply:
I think these commits are really close to be something really useful.
It might help to split this PR to smaller ones.

I think the absolute minimum is `ForceStyleColorSpaceConversion()` and the conversion functions it needs. IMHO it would be better if we did not call it 'force' if it is not aware of the current color space. Instead, if we call it something like ConvertFromSRGBToLinear(), then it is slightly more visible to the user that the function just does a color conversion blindly, and user better know.

I would be very happy if there was a PR which only provided this utility function and nothing else. It would break nothing, and would have value of allowing people to get correct rendering more easily. The other fixes can be done later as separate PR.